src/app/shared/services/snackbar.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import {
+  MatSnackBar,
+  MatSnackBarConfig,
+  MatSnackBarHorizontalPosition,
+  MatSnackBarVerticalPosition,
+} from '@angular/material/snack-bar';
+import { SnackAlertComponent } from '../components/snackbar/snackbar.component';
+import { ALERT_TYPES } from '../models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SnackbarService {
+  private _horizontalPosition: MatSnackBarHorizontalPosition = 'center';
+  private _verticalPosition: MatSnackBarVerticalPosition = 'bottom';
+  private _actionButtonLabel = 'OK';
+  private _action = false;
+  private _setAutoHide = true;
+  private _autoHide = 5000;
```

Comment:
TS already ensures those are private, do we need these `_` prefixes?

Reply:
It is just a way to clarify the difference between the local variable author and the property referencing the local variable value of author. The underscore character is just another valid character which can be used in variable names and/or property names.

https://www.typescripttutorial.net/typescript-tutorial/typescript-getters-setters/
https://stackoverflow.com/posts/47146127/timeline)

You need to have a variable that has a different name then the getter/setter. The most common way is to prefix the variable with an _.

Otherwise if you don't do this you will have an endless loop of calls to the getter/setter.

class A {
    constructor() {
        this._a = '';
    }

    get a() {
        // If we don't call it like this
        // We will be calling the getter again
        // Which in return will call itself again
        return this._a
    }
}

Reply:
Jak to jest z tym private? https://www.dotnettricks.com/learn/typescript/access-modifiers
Public - By default, members (properties and methods) of the TypeScript class are public - so you don’t need to prefix members with the public keyword. Public members are accessible everywhere without restrictions even from the multiple level sub-classes without any compile errors.

Reply:
OK, ale IMO gettery i settery mają sens głównie wtedy, gdy chcesz dorzucić jakąś logikę. Nie widzę sensu w prywatnych polach prefiksowanych `_`, do których dopisuje się trywialne settery. Czemu po prostu nie zdefiniować tych pól jako publiczne?

Reply:
> Jak to jest z tym private? dotnettricks.com/learn/typescript/access-modifiers

A z czym masz wątpliwości?

Reply:
robie tak samo jak @mgol, stosuje setter i gettery jeśli przykładowo chce sprawdzić coś na wejściu i zwrócić coś innego na wyjściu. W innych przypadkach nie stosuje